doc/exts/xref.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+
+from docutils import nodes
+
+from sphinx.util import caption_ref_re
+
+
+def xref(typ, rawtext, text, lineno, inliner, options={}, content=[]):
```

Comment:
Nothing like this already existed (weird), or have you vendored something?

Reply:
I used my time machine and tried to explain this in 906d220. But TL;DR: Yes, I vendored it in because it wasn't built in, and not available on PyPI.